### PR TITLE
First batch of improvement based on issue #1

### LINKED ISSUE #1: first issue
I know this repository is still in a Work In Progress state, but I create an issue to list my first questions/stuff to add, not to forget them.
I've just taken a very quick look at the project, so maybe my observation are dumb. We can split this in several issues if you are ok with it.

## Stuff to add
* readme :stuck_out_tongue_winking_eye: 
* CI
* config.toml (at least an example, but I'm not sure there are some secrets in it).
* tests :roll_eyes: 

## improvements
* the cli seems only to output the producer to get it's it right ? it'm not a fan on this API

* in wikibase/mod.rs there are some structures, whereas there are also some in wikibase/structures.rs. maybe we can move the structures in mod.rs or create an additional mod for them if they are different.

* really minor, but I prefer for the `new` method to be part of the class, but the module (in `sparql_client.rs`, `client.rs` and `mod.rs`
## Questions
the file `action/import_gtfs.rs` is empty, it's because you plan to move the code from `Client::import_lines` in it?

### PR DESCRIPTION
**Pull request by Tristramg**
Closes #1 (except tests, but I’ll do a specific issue for that)

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+This program is a toolkit to populate that database from a [GTFS](https://gtfs.org) file.
+
+It allows to populate missing features in the database.
+
+The tool is designed to be idempotent: importing twice the same file, or different files from the same producer won’t generate any duplicate.
+
+## Installation
+
+Transit topo tools is written in [Rust](https://www.rust-lang.org/).
```
Comment (antoine-de) [inline readme.md]: **are**

File: readme.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Configuration
+
+You need to configure `config.toml` in order to set which are the ids of basic identifiers (“bus”…) or property (“gtfs_id”).
+
+See `config.example.toml` how to set it up.
+
+## Running
+
+Identifiers can of entities can be the same across different producers. That is why we require to set a producer with the `--producer` (or `-p`) flag.
```
Comment (antoine-de) [inline readme.md]: s/can//

File: readme.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+For instance, to import all the lines of the producer `Q4` from the local file `gtfs.zip`, run:
+
+    cargo run --relase -- -p Q4 -i gtfs.zip
```
Comment (antoine-de) [inline readme.md]: release

File: src/wikibase/client.rs [hunk 4]
```diff
@@ -11,15 +11,15 @@ pub struct Client {
     token: Option<String>,
 }
 
-pub fn new(config: super::Config) -> Client {
-    Client {
-        client: reqwest::Client::new(),
-        config,
-        token: None,
+impl Client {
+    pub fn new(config: super::Config) -> Self {
```
Comment (antoine-de) [inline src/wikibase/client.rs]: it's really minor, but I'm not a fan of using contextual import, I prefer the full name to `super`

File: src/wikibase/mod.rs [hunk 5]
```diff
@@ -46,19 +46,18 @@ impl Config {
...
-    let config = toml::from_str::<Config>(&content)?;
-
-    Ok(Client {
-        api: client::new(config.clone()),
-        sparql: sparql_client::new(config),
-    })
-}
-
 impl Client {
+    pub fn new(config_file: &str) -> Result<Self, Error> {
```
Comment (antoine-de) [inline src/wikibase/mod.rs]: completely minor too: you could have made it a bit more generic:
```rust
pub fn new<P: AsRef<Path>>(config_file: P) -> Result<Self, Error> {
}
```
(it's a bit more idiomatic)

**Comment by Tristramg** [comment]
updated with the various remarks

**Pull request merged**